sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/MetricsAdvisorKeyCredentialTests.cs

```diff
@@ -0,0 +1,99 @@
+        public async Task MetricsAdvisorKeyCredentialSendsSecretInMetricsAdvisorClient()
+        {
+            MockResponse response = new MockResponse(200);
+            response.SetContent("{}");
+
+            MockTransport mockTransport = new MockTransport(response);
+
+            string expectedSubscriptionKey = "fakeSubscriptionKey";
+            string expectedApiKey = "fakeApiKey";
+            MetricsAdvisorKeyCredential credential = new MetricsAdvisorKeyCredential(expectedSubscriptionKey, expectedApiKey);
+
+            MetricsAdvisorClient client = CreateInstrumentedClient(mockTransport, credential);
+
+            IAsyncEnumerator<MetricFeedback> asyncEnumerator = client.GetAllFeedbackAsync(FakeGuid).GetAsyncEnumerator();
+            await asyncEnumerator.MoveNextAsync();
+
+            MockRequest request = mockTransport.Requests.First();
+
+            Assert.That(request.Headers.TryGetValue(Constants.SubscriptionAuthorizationHeader, out string subscriptionKey));
+            Assert.That(request.Headers.TryGetValue(Constants.ApiAuthorizationHeader, out string apiKey));
+
+            Assert.That(subscriptionKey, Is.EqualTo(expectedSubscriptionKey));
+            Assert.That(apiKey, Is.EqualTo(expectedApiKey));
+        }
+
+        [Test]
+        public async Task MetricsAdvisorKeyCredentialSendsSecretInMetricsAdvisorAdministrationClient()
+        {
+            MockResponse response = new MockResponse(204);
+            MockTransport mockTransport = new MockTransport(response);
+
+            string expectedSubscriptionKey = "fakeSubscriptionKey";
+            string expectedApiKey = "fakeApiKey";
+            MetricsAdvisorKeyCredential credential = new MetricsAdvisorKeyCredential(expectedSubscriptionKey, expectedApiKey);
+
+            MetricsAdvisorAdministrationClient adminClient = CreateInstrumentedAdministrationClient(mockTransport, credential);
+
+            await adminClient.DeleteAlertConfigurationAsync(FakeGuid);
+
+            MockRequest request = mockTransport.Requests.First();
+
+            Assert.That(request.Headers.TryGetValue(Constants.SubscriptionAuthorizationHeader, out string subscriptionKey));
+            Assert.That(request.Headers.TryGetValue(Constants.ApiAuthorizationHeader, out string apiKey));
+
+            Assert.That(subscriptionKey, Is.EqualTo(expectedSubscriptionKey));
+            Assert.That(apiKey, Is.EqualTo(expectedApiKey));
+        }
+
+        [Test]
+        public async Task MetricsAdvisorKeyCredentialUpdatesSecret()
+        {
+            MockResponse response = new MockResponse(204);
+            MockTransport mockTransport = new MockTransport(response);
+            MetricsAdvisorKeyCredential credential = new MetricsAdvisorKeyCredential("fakeSubscriptionKey", "fakeApiKey");
+
+            string expectedSubscriptionKey = "newFakeSubscriptionKey";
+            string expectedApiKey = "newFakeApiKey";
+
+            credential.Update(expectedSubscriptionKey, expectedApiKey);
```

Comment:
because of mocking, I believe that you are attached to the client you created. A user will use the update method in an already live client insteadcreating one with the updated credentials. 
Do you think it is worth adding a live test that reflects that behavior?

Reply:
We can move the `Update` call to the point after creating the client.